src/Workspaces/Core/MSBuild/MSBuild/MSBuildWorkspace.cs

```diff
@@ -382,7 +380,24 @@ protected override void ApplyDocumentTextChanged(DocumentId documentId, SourceTe
             }
         }
 
-        private static Encoding? DetermineEncoding(SourceText text, Document document)
+        protected override void ApplyAdditionalDocumentTextChanged(DocumentId documentId, SourceText text)
+        {
+            var document = this.CurrentSolution.GetAdditionalDocument(documentId);
+            if (document != null)
+            {
+                var encoding = DetermineEncoding(text, document);
+                if (document.FilePath is null)
+                {
+                    var message = string.Format(WorkspaceMSBuildResources.Path_for_document_0_was_null, document.Name);
```

Comment:
nit: Do we need this to say `Path for additional document {0} was null` ?

Reply:
Yeah that would probably be preferable.